File: frontend/src/app/services/exam.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class ExamService {

  constructor(private httpClient : HttpClient) { }

  public listExams()
  {
    return this.httpClient.get(`${baseUrl}/api/v1/exams/`);
  }

  public createExam(exam : any)
  {
    return this.httpClient.post(`${baseUrl}/api/v1/exams/`, exam);
  }

  public deleteExam(id : any)
  {
    return this.httpClient.delete(`${baseUrl}/api/v1/exams/${id}`)
  }

  public getExamById(id : any)
  {
    return this.httpClient.get(`${baseUrl}/api/v1/exams/${id}`);
  }

  public updateExam(exam : any)
  {
    return this.httpClient.put(`${baseUrl}/api/v1/exams/`, exam);
  }

  public getExamByCategory(categoryId : any)
  {
    return this.httpClient.get(`${baseUrl}/api/v1/exams/category/${categoryId}`);
  }

  public getExamsActive()
  {
    return this.httpClient.get(`${baseUrl}/api/v1/exams/active`);
  }

  public getExamsActiveByCategory(categoryId : any)
  {
    return this.httpClient.get(`${baseUrl}/api/v1/exams/category/active/${categoryId}`);
  }
}

File: backend/src/main/java/com/sistema/examenes/sistema_examenes_backend/services/ExamService.java
package com.sistema.examenes.sistema_examenes_backend.services;

import java.util.List;
import java.util.Set;

import com.sistema.examenes.sistema_examenes_backend.models.Category;
import com.sistema.examenes.sistema_examenes_backend.models.Exam;

public interface ExamService
{
	Exam createExam(Exam exam);
	
	Exam updateExam(Exam exam);
	
	Set<Exam> getExams();
	
	Exam getExamById(Long id);
	
	void deleteExam(Long id);
	
	List<Exam> getExamByCategory(Category category);
	
	List<Exam> getExamsActive();
	
	List<Exam> getExamsActiveByCategory(Category category);
}